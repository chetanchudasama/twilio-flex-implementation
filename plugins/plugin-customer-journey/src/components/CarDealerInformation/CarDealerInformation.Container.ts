import { connect } from "react-redux";
import { Dispatch } from "redux";
import { getWorkerIsAvailableForOutboundCall } from "../../states/selectors";
import { AppState } from "../../states";
import { DispatchToProps, StateToProps } from "./CarDealerInformation.Props";
import CarDealerInformation from "./CarDealerInformation";

const mapStateToProps = (state: AppState): StateToProps => {
  return {
    canMakeCall: getWorkerIsAvailableForOutboundCall(state),
  };
};

const mapDispatchToProps = (dispatch: Dispatch<any>): DispatchToProps => ({});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CarDealerInformation);
