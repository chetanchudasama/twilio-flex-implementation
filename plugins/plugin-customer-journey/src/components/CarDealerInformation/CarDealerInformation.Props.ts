export interface StateToProps {
  canMakeCall: boolean;
}

export interface DispatchToProps {}
