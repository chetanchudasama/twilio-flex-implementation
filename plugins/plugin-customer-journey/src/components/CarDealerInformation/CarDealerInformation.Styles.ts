import { secondary, black } from "@common/components";
import styled from "react-emotion";

export const CarDealerInformationStyles = styled("div")({
  marginTop: 8,
  ".dealer-name": {
    fontSize: "16px",
    fontWeight: 600,
    paddingBottom: "8px",
    marginLeft: 5,
  },
  ".info-icon": {
    color: secondary,
    opacity: 0.4,
  },
  ".dealer-info": {
    fontSize: "14px",
    padding: "2px 10px 3px 10px",
    wordBreak: "break-word",
  },
  ".dealer-info-url": {
    fontSize: "14px",
    wordBreak: "break-word",
    padding: "2px 0px 3px 10px",
  },
  ".dealer-info-url a": {
    color: black,
  },
  ".dealer-phone, dealer-address": {
    display: "-webkit-box",
    WebkitBoxOrient: "vertical",
    WebkitLineClamp: 2,
    overflow: "hidden",
    textOverflow: "ellipsis",
  },
});
