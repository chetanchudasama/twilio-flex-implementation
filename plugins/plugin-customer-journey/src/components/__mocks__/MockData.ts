import { WorkerAttributes } from "@twilio/flex-ui";
import { namespace } from "../../states";

const workerAttributes: WorkerAttributes = {
  contact_uri: "client:test",
  email: "test@test.com",
  full_name: "test user",
  image_url: "img_url",
  roles: ["admin"],
  agentId: 1,
};

const conferenceSource = {
  source: {
    participants: [
      {
        callSid: "callSid",
        participantType: "customer",
        workerSid: "",
      },
      {
        callSid: "callSid",
        participantType: "worker",
        workerSid: "workerSid",
      },
    ],
  },
};

const conferenceState = new Map();
conferenceState.set("taskSid", conferenceSource);

export const flexManagerMock = jest.fn().mockReturnValue({
  workerClient: {
    source: {},
    sid: "workerSid",
    name: "test user",
    fullName: "test user",
    activityName: "Offline",
    attributes: workerAttributes,
    dateUpdated: new Date(),
    isAvailable: false,
    activityDuration: "",
    dateStatusChanged: new Date(1).toISOString(),
    activity: {
      available: false,
      name: "Offline",
      sid: "OfflineSid",
    },
  },
  store: {
    getState: jest.fn().mockReturnValue({
      flex: {
        worker: {
          attributes: workerAttributes,
        },
        session: {
          ssoTokenPayload: {
            token: "ssoToken",
          },
        },
        conferences: {
          states: conferenceState,
        },
      },
      [namespace]: {
        phoneNumberList: {
          phoneNumberList: [
            {
              friendlyName: "Sales",
              phoneNumber: "+1234567891",
              queueSid: "QUEUE_SID",
            },
          ],
        },
      },
    }),
  },
  strings: {
    DIALPADExternalTransferSearchInputLabel: "Search By Name or PhoneNumber",
    DIALPADExternalTransferHoverOver: "Add",
    DIALPADExternalTransferPopupHeader: "External Transfer",
  },
  serviceConfiguration: {
    outbound_call_flows: {
      default: {
        caller_id: "CALLER_ID",
      },
    },
  },
});
