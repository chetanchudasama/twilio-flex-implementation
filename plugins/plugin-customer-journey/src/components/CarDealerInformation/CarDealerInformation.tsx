import { Grid } from "@material-ui/core";
import LanguageIcon from "@material-ui/icons/Language";
import LocationOnIcon from "@material-ui/icons/LocationOn";
import StarIcon from "@material-ui/icons/Star";
import { Actions, IconButton } from "@twilio/flex-ui";
import React from "react";
import { DealerResponseModel } from "../../models/DealerResponseModel";
import { StateToProps, DispatchToProps } from "./CarDealerInformation.Props";
import { CarDealerInformationStyles } from "./CarDealerInformation.Styles";

export interface CarDealerInformationProp {
  dealerInfo: DealerResponseModel;
  distance?: number;
}

export type Props = StateToProps & DispatchToProps & CarDealerInformationProp;

const CarDealerInformation: React.FC<Props> = ({
  dealerInfo,
  distance,
  canMakeCall,
}) => {
  const triggerOutboundCall = () => {
    if (canMakeCall && dealerInfo.phone) {
      const payload = {
        destination: dealerInfo.phone,
      };
      Actions.invokeAction("StartOutboundCall", payload);
    }
  };

  return (
    <CarDealerInformationStyles>
      <Grid container>
        <Grid item xs={12} className="dealer-name">
          {dealerInfo.dealerName}
        </Grid>
        <Grid item container>
          <Grid item xs={4}>
            <Grid container>
              <Grid item xs={1}>
                <LocationOnIcon fontSize="small" className="info-icon" />
              </Grid>
              <Grid item xs={11} className="dealer-info dealer-address">
                {dealerInfo.town + (distance ? ` (${distance} miles)` : "")}
              </Grid>
            </Grid>
          </Grid>
          <Grid item xs={2}>
            <Grid container>
              <Grid item xs={2}>
                <StarIcon fontSize="small" className="info-icon" />
              </Grid>
              <Grid item xs={10} className="dealer-info dealer-rating">
                {dealerInfo.rating ?? " -"}
              </Grid>
            </Grid>
          </Grid>
          <Grid item xs={4}>
            <Grid container>
              <Grid item xs={1}>
                <IconButton
                  icon="Call"
                  className="info-icon"
                  disabled={!canMakeCall}
                  onClick={() => triggerOutboundCall()}
                />
              </Grid>
              <Grid item xs={11} className="dealer-info dealer-phone">
                {dealerInfo.phone}
              </Grid>
            </Grid>
          </Grid>
          <Grid item xs={2}>
            <Grid container>
              <Grid item xs={2}>
                <LanguageIcon fontSize="small" className="info-icon" />
              </Grid>
              <Grid item xs={10} className="dealer-info-url">
                <a href={dealerInfo.url} target="blank">
                  Website
                </a>
              </Grid>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </CarDealerInformationStyles>
  );
};

export default CarDealerInformation;
