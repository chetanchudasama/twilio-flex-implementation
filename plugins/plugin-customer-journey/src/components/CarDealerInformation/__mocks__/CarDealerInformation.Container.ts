import CarDealerInformation from "./CarDealerInformation";

export default CarDealerInformation;
