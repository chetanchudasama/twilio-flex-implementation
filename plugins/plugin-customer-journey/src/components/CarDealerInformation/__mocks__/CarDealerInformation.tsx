import React from "react";
import { Props } from "../CarDealerInformation";

const CarDealerInformation: React.FC<Props> = (props) => <div />;

export default CarDealerInformation;
