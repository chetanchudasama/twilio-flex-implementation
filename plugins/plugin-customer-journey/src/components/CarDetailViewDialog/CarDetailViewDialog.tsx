import { Shared, ResponsiveDialog } from "@common/components";
import React, { useEffect, useState, useMemo } from "react";
import { Manager } from "@twilio/flex-ui";
import ExpandMoreRoundedIcon from "@material-ui/icons/ExpandMoreRounded";
import ExpandLessRoundedIcon from "@material-ui/icons/ExpandLessRounded";
import { Grid, Button } from "@material-ui/core";
import { v4 } from "uuid";
import { CarDetailViewDialogStyles } from "./CarDetailViewDialog.Styles";
import { VehicleDetailModel } from "../../models/VehicleDetailModel";
import CarDealerInformation from "../CarDealerInformation/CarDealerInformation.Container";
import { useVehicleService } from "../../services/vehicle.service";
import { AppState, namespace } from "../../states";
import { showErrorMessage } from "../../Notifications";
import { StateToProps, DispatchToProps } from "./CarDetailViewDialog.Props";
import Loading from "../LoadingComponent/Loading";
import { VehicleImage } from "../VehicleImage/VehicleImage";

export interface CarDetailViewDialogProps {
  open: boolean;
  vehicleId: number;
  postCode: string;
  handleCloseDialog: () => void;
}

type Props = StateToProps & DispatchToProps & CarDetailViewDialogProps;

const CarDetailViewDialog: React.FC<Props> = ({
  open,
  vehicleId,
  postCode,
  handleCloseDialog,
}) => {
  const state: AppState = Manager.getInstance().store.getState();
  const token: string = useMemo(() => {
    return state.flex.session.ssoTokenPayload.token ?? "";
  }, [state.flex.session.ssoTokenPayload]);

  const phoenixToken: string = useMemo(() => {
    return state.crm?.crmState?.phoenixToken ?? "";
  }, [state.crm?.crmState?.phoenixToken]);

  const vehicleService = useVehicleService(token, phoenixToken);

  const [vehicleDetail, setVehicleDetail] = useState(new VehicleDetailModel());
  const [showAllSpecification, setShowAllSpecification] = useState(false);
  const [showFullDescription, setShowFullDescription] = useState(false);
  const [isLoading, setIsLoading] = useState<boolean>(false);

  useEffect(() => {
    const getVehicleDetail = async () => {
      setIsLoading(true);
      vehicleService
        .getVehicleDetail(vehicleId, postCode)
        .then((response) => {
          setVehicleDetail(response);
          setIsLoading(false);
        })
        .catch((error) => {
          showErrorMessage(
            "Error loading vehicle detail, please try again!",
            "",
            true
          );
          handleCloseDialog();
        });
    };
    getVehicleDetail();
    // TODO come back to this and see if we can fix
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const vehicleSpecificationList = [
    {
      propertyName: "Transmission",
      propertyValue: vehicleDetail.trans ?? "-",
    },
    {
      propertyName: "Body Type",
      propertyValue: vehicleDetail.body ?? "-",
    },
    {
      propertyName: "Engine Size",
      propertyValue: `${vehicleDetail.engine ?? "0"}L`,
    },
    {
      propertyName: "Doors",
      propertyValue: `${vehicleDetail.doorsCount ?? "0"}dr`,
    },
    {
      propertyName: "Fuel Type",
      propertyValue: vehicleDetail.fuel ?? "-",
    },
    {
      propertyName: "Colour",
      propertyValue: vehicleDetail.colour ?? "-",
    },
    {
      propertyName: "Trim",
      propertyValue: vehicleDetail.trim ?? "-",
    },
    {
      propertyName: "Year",
      propertyValue: vehicleDetail.year ?? "-",
    },
    {
      propertyName: "Mileage",
      propertyValue: `${vehicleDetail.mileage ?? "0"} mi`,
    },
    {
      propertyName: "Distance",
      propertyValue: vehicleDetail.distance ?? "0",
    },
    {
      propertyName: "Reg. Number",
      propertyValue: vehicleDetail.regNumber ?? "-",
    },
  ];

  const toggleCarSpecification = () => {
    setShowAllSpecification(!showAllSpecification);
  };

  const toggleCarDescription = () => {
    setShowFullDescription(!showFullDescription);
  };

  return (
    <>
      <ResponsiveDialog
        open={open}
        onCancel={handleCloseDialog}
        onConfirm={() => {}}
        maxWidth="lg"
        showActionButtons={false}
        titleCloseIcon
      >
        <CarDetailViewDialogStyles>
          {isLoading && <Loading />}
          {!isLoading && (
            <Grid container spacing={8}>
              <Grid item xs={6}>
                <Grid container>
                  <Grid item xs={12} className="car-dialog-image">
                    <VehicleImage
                      image={vehicleDetail.images[0]}
                      imageList={vehicleDetail.images}
                      imagesCount={vehicleDetail.images.length}
                      showImageCount
                    />
                  </Grid>
                  <Grid item xs={12} className="dealer-info-content">
                    <CarDealerInformation
                      dealerInfo={vehicleDetail.dealer}
                      distance={vehicleDetail.distance!}
                    />
                  </Grid>
                </Grid>
              </Grid>
              <Grid item xs={6}>
                <Grid container className="car-spec-detail">
                  <Grid item xs={12}>
                    <div className="car-price">
                      {Shared.getFormattedCurrencyValue(vehicleDetail.price)}
                    </div>
                  </Grid>
                  <Grid item xs={12}>
                    <ul className="car-specification">
                      <li>{vehicleDetail.make}</li>
                      <li>{vehicleDetail.model}</li>
                      <li>{vehicleDetail.body}</li>
                      <li>{vehicleDetail.doorsCount}dr</li>
                    </ul>
                  </Grid>
                  {vehicleSpecificationList.map((vehicleInfo, index) => (
                    <React.Fragment key={v4()}>
                      <Grid
                        item
                        xs={6}
                        className={!(index % 2) ? "space-right" : "space-left"}
                      >
                        <Grid container className="car-prop" spacing={8}>
                          <Grid item xs={5} className="car-prop-type">
                            {vehicleInfo.propertyName}
                          </Grid>
                          <Grid item xs={7} className="car-prop-value">
                            {vehicleInfo.propertyValue}
                          </Grid>
                        </Grid>
                      </Grid>
                    </React.Fragment>
                  ))}
                  <Grid item xs={6} />
                  {showAllSpecification &&
                    vehicleDetail.specs &&
                    vehicleDetail.specs.map((specDetail, index) => (
                      <React.Fragment key={v4()}>
                        <Grid
                          item
                          xs={6}
                          className={
                            !(index % 2) ? "space-right" : "space-left"
                          }
                        >
                          <Grid container className="car-prop" spacing={8}>
                            <Grid item xs={5} className="car-prop-type">
                              {specDetail.name}
                            </Grid>
                            <Grid item xs={7} className="car-prop-value">
                              {specDetail.desc}
                            </Grid>
                          </Grid>
                        </Grid>
                      </React.Fragment>
                    ))}
                  {vehicleDetail.specs && vehicleDetail.specs.length > 0 && (
                    <Grid item xs={12} className="expand-more">
                      <Button
                        onClick={() => toggleCarSpecification()}
                        className="expand-more-link"
                        disableRipple
                      >
                        <div>
                          {showAllSpecification
                            ? "View less specs"
                            : "View full specs"}
                        </div>
                        {showAllSpecification ? (
                          <ExpandLessRoundedIcon />
                        ) : (
                          <ExpandMoreRoundedIcon />
                        )}
                      </Button>
                    </Grid>
                  )}
                  {vehicleDetail.options && (
                    <>
                      <Grid item xs={12} className="car-description-heading">
                        Description
                      </Grid>
                      <Grid item xs={12}>
                        <div
                          id="car-desc-text"
                          className={`car-description ${
                            showFullDescription
                              ? "car-description-expanded"
                              : ""
                          }`}
                        >
                          {vehicleDetail.options}
                        </div>
                      </Grid>
                      <Grid item xs={12} className="expand-more">
                        <Button
                          onClick={() => toggleCarDescription()}
                          className="expand-more-link"
                          disableRipple
                        >
                          <div>
                            {showFullDescription
                              ? "Read less description"
                              : "Read full description"}
                          </div>
                          {showFullDescription ? (
                            <ExpandLessRoundedIcon />
                          ) : (
                            <ExpandMoreRoundedIcon />
                          )}
                        </Button>
                      </Grid>
                    </>
                  )}
                </Grid>
              </Grid>
            </Grid>
          )}
        </CarDetailViewDialogStyles>
      </ResponsiveDialog>
    </>
  );
};

export default CarDetailViewDialog;
