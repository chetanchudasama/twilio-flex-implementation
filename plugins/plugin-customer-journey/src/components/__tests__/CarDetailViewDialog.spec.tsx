import React from "react";
import { mount } from "enzyme";
import { create, act, ReactTestRenderer } from "react-test-renderer";
import { act as Act } from "react-dom/test-utils";
import { Manager } from "@twilio/flex-ui";
import { flexManagerMock } from "../__mocks__/MockData";
import CarDetailViewDialog, {
  CarDetailViewDialogProps,
} from "../CarDetailViewDialog/CarDetailViewDialog";
import {
  StateToProps,
  DispatchToProps,
} from "../CarDetailViewDialog/CarDetailViewDialog.Props";
import { VehicleDetailModel } from "../../models/VehicleDetailModel";
import { showErrorMessage } from "../../Notifications";
import CarDealerInformation from "../CarDealerInformation/CarDealerInformation.Container";
import { DealerResponseModel } from "../../models/DealerResponseModel";

type Props = StateToProps & DispatchToProps & CarDetailViewDialogProps;

const props: Props = {
  open: true,
  handleCloseDialog: jest.fn(() => {}),
  vehicleId: 901,
  postCode: "Postcode",
};

Manager.getInstance = flexManagerMock;

jest.mock("../../Notifications", () => {
  return {
    showErrorMessage: jest.fn(() => {}),
  };
});

jest.mock("../../services/vehicle.service");

const mockGetVehicleDetail = jest.fn();

jest.mock("../../services/vehicle.service", () => ({
  useVehicleService: jest.fn(() => ({
    getVehicleDetail: mockGetVehicleDetail,
  })),
}));

jest.mock("@common/components", () => {
  const components = jest.requireActual("@common/components");
  return {
    ...components,
    ResponsiveDialog: jest.fn(({ children }) => children),
    Shared: {
      getFormattedCurrencyValue: jest.fn((value) => {
        return value;
      }),
      getFormattedText: jest.fn((value) => {
        return value.join("| ");
      }),
    },
  };
});

jest.mock("../CarDealerInformation/CarDealerInformation.Container");
jest.mock("../CarDealerInformation/CarDealerInformation");

const vehicleDetail = Object.assign(new VehicleDetailModel(), {
  vehicleId: "901",
  dealerAdminFee: 200,
  image: "ImageUrl",
  images: ["ImageUrl"],
  make: "Bentley Motors",
  model: "Bentley Continental GT",
  price: 500000,
  mileage: 13,
  regNumber: "BM12853",
  regYear: 2016,
  year: 2016,
  trans: "Automatic",
  fuel: "Petrol",
  body: "Convertible",
  colour: "white",
  dealer: Object.assign(new DealerResponseModel(), {
    dealerName: "MR john",
    county: "UK",
    building: null,
    streetName: "46 Grenoble Road",
    town: "BREADSTONE",
    rating: 4.5,
    phone: "+447700870088",
    url: "https://namesbee.com/car-dealership-names/",
    postcode: "GL13 1XD",
  }),
  specs: [
    {
      name: "BI-OCULAR-NC3000",
      desc: "Night Vision Single Tube Bi-ocular",
    },
  ],
});

const GetVehicleDetail = () => {
  mockGetVehicleDetail.mockImplementationOnce(() =>
    Promise.resolve(vehicleDetail)
  );
};

jest.mock("../VehicleImage/VehicleImage", () => {
  return {
    __esModule: true,
    VehicleImage: () => {
      return <div className="vehicleImage">Vehicle Image</div>;
    },
  };
});

describe("CarDetailViewDialog", () => {
  it("renders correctly on load", async () => {
    GetVehicleDetail();
    let wrapper: ReactTestRenderer;
    await act(async () => {
      wrapper = create(<CarDetailViewDialog {...props} />);
    });

    expect(mockGetVehicleDetail).toHaveBeenCalled();
    expect(wrapper!.toJSON()).toMatchSnapshot();
  });

  it("should call getVehicleDetail API, when component is rendered", async () => {
    GetVehicleDetail();
    const wrapper = mount(<CarDetailViewDialog {...props} />);

    expect(wrapper.find(".progress").exists()).toBeTruthy();
    await Act(async () => {
      expect(mockGetVehicleDetail).toHaveBeenCalled();
    });
    wrapper.update();

    expect(wrapper.find(".progress").exists()).toBeFalsy();
    expect(wrapper.find(".car-dialog-image").exists()).toBeTruthy();
  });

  it("Error message should be displayed, when should call getVehicleDetail API returns an error", async () => {
    mockGetVehicleDetail.mockImplementationOnce(() =>
      Promise.reject({ statusCode: 500 })
    );
    mount(<CarDetailViewDialog {...props} />);

    await Act(async () => {
      expect(mockGetVehicleDetail).toHaveBeenCalled();
    });
    expect(showErrorMessage).toHaveBeenCalledWith(
      "Error loading vehicle detail, please try again!",
      "",
      true
    );
    expect(props.handleCloseDialog).toHaveBeenCalled();
  });

  it("should render VehicleImage and CarDealerInformation components", async () => {
    GetVehicleDetail();
    const wrapper = mount(<CarDetailViewDialog {...props} />);

    await Act(async () => {
      expect(mockGetVehicleDetail).toHaveBeenCalled();
    });
    wrapper.update();

    expect(wrapper.find(".vehicleImage").exists()).toBeTruthy();
    expect(wrapper.find(CarDealerInformation).exists()).toBeTruthy();
  });
});
