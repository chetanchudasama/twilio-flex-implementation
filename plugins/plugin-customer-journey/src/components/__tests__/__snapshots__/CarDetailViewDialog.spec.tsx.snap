// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarDetailViewDialog renders correctly on load 1`] = `
<div
  className="css-k8vt34"
>
  <div
    className="MuiGrid-container-108 MuiGrid-spacing-xs-8-130"
  >
    <div
      className="MuiGrid-item-109 MuiGrid-grid-xs-6-142"
    >
      <div
        className="MuiGrid-container-108"
      >
        <div
          className="MuiGrid-item-109 MuiGrid-grid-xs-12-148 car-dialog-image"
        >
          <div
            className="vehicleImage"
          >
            Vehicle Image
          </div>
        </div>
        <div
          className="MuiGrid-item-109 MuiGrid-grid-xs-12-148 dealer-info-content"
        >
          <div />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-109 MuiGrid-grid-xs-6-142"
    >
      <div
        className="MuiGrid-container-108 car-spec-detail"
      >
        <div
          className="MuiGrid-item-109 MuiGrid-grid-xs-12-148"
        >
          <div
            className="car-price"
          >
            500000
          </div>
        </div>
        <div
          className="MuiGrid-item-109 MuiGrid-grid-xs-12-148"
        >
          <ul
            className="car-specification"
          >
            <li>
              Bentley Motors
            </li>
            <li>
              Bentley Continental GT
            </li>
            <li>
              Convertible
            </li>
            <li>
              dr
            </li>
          </ul>
        </div>
        <div
          className="MuiGrid-item-109 MuiGrid-grid-xs-6-142 space-right"
        >
          <div
            className="MuiGrid-container-108 MuiGrid-spacing-xs-8-130 car-prop"
          >
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-5-141 car-prop-type"
            >
              Transmission
            </div>
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-7-143 car-prop-value"
            >
              Automatic
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-109 MuiGrid-grid-xs-6-142 space-left"
        >
          <div
            className="MuiGrid-container-108 MuiGrid-spacing-xs-8-130 car-prop"
          >
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-5-141 car-prop-type"
            >
              Body Type
            </div>
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-7-143 car-prop-value"
            >
              Convertible
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-109 MuiGrid-grid-xs-6-142 space-right"
        >
          <div
            className="MuiGrid-container-108 MuiGrid-spacing-xs-8-130 car-prop"
          >
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-5-141 car-prop-type"
            >
              Engine Size
            </div>
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-7-143 car-prop-value"
            >
              0L
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-109 MuiGrid-grid-xs-6-142 space-left"
        >
          <div
            className="MuiGrid-container-108 MuiGrid-spacing-xs-8-130 car-prop"
          >
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-5-141 car-prop-type"
            >
              Doors
            </div>
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-7-143 car-prop-value"
            >
              0dr
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-109 MuiGrid-grid-xs-6-142 space-right"
        >
          <div
            className="MuiGrid-container-108 MuiGrid-spacing-xs-8-130 car-prop"
          >
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-5-141 car-prop-type"
            >
              Fuel Type
            </div>
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-7-143 car-prop-value"
            >
              Petrol
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-109 MuiGrid-grid-xs-6-142 space-left"
        >
          <div
            className="MuiGrid-container-108 MuiGrid-spacing-xs-8-130 car-prop"
          >
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-5-141 car-prop-type"
            >
              Colour
            </div>
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-7-143 car-prop-value"
            >
              white
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-109 MuiGrid-grid-xs-6-142 space-right"
        >
          <div
            className="MuiGrid-container-108 MuiGrid-spacing-xs-8-130 car-prop"
          >
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-5-141 car-prop-type"
            >
              Trim
            </div>
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-7-143 car-prop-value"
            >
              
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-109 MuiGrid-grid-xs-6-142 space-left"
        >
          <div
            className="MuiGrid-container-108 MuiGrid-spacing-xs-8-130 car-prop"
          >
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-5-141 car-prop-type"
            >
              Year
            </div>
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-7-143 car-prop-value"
            >
              2016
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-109 MuiGrid-grid-xs-6-142 space-right"
        >
          <div
            className="MuiGrid-container-108 MuiGrid-spacing-xs-8-130 car-prop"
          >
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-5-141 car-prop-type"
            >
              Mileage
            </div>
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-7-143 car-prop-value"
            >
              13 mi
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-109 MuiGrid-grid-xs-6-142 space-left"
        >
          <div
            className="MuiGrid-container-108 MuiGrid-spacing-xs-8-130 car-prop"
          >
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-5-141 car-prop-type"
            >
              Distance
            </div>
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-7-143 car-prop-value"
            >
              0
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-109 MuiGrid-grid-xs-6-142 space-right"
        >
          <div
            className="MuiGrid-container-108 MuiGrid-spacing-xs-8-130 car-prop"
          >
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-5-141 car-prop-type"
            >
              Reg. Number
            </div>
            <div
              className="MuiGrid-item-109 MuiGrid-grid-xs-7-143 car-prop-value"
            >
              BM12853
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-109 MuiGrid-grid-xs-6-142"
        />
        <div
          className="MuiGrid-item-109 MuiGrid-grid-xs-12-148 expand-more"
        >
          <button
            className="MuiButtonBase-root-231 MuiButton-root-205 MuiButton-text-207 MuiButton-flat-210 expand-more-link"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-206"
            >
              <div>
                View full specs
              </div>
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-234"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M24 24H0V0h24v24z"
                  fill="none"
                  opacity=".87"
                />
                <path
                  d="M15.88 9.29L12 13.17 8.12 9.29a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l4.59 4.59c.39.39 1.02.39 1.41 0l4.59-4.59c.39-.39.39-1.02 0-1.41-.39-.38-1.03-.39-1.42 0z"
                />
              </svg>
            </span>
          </button>
        </div>
        
      </div>
    </div>
  </div>
</div>
`;
